### PR TITLE
better handling of errors + handle bad refreshToken

### PR DESCRIPTION
**Pull request by badaz**
Sending a refresh token with bad scopes returns a 400, handle this case to throw an AccessDeniedError in this case

### CODE REVIEW

File: src/Error.js [hunk 1]
```diff
@@ -63,22 +63,38 @@ InternalServerError.prototype = Object.create(HttpError.prototype);
 InternalServerError.prototype.constructor = InternalServerError;
 
 function handleBadResponse(response) {
+  let error;
```
Comment (jdeniau) [inline src/Error.js]: Make another PR with this file ?

File: src/Error.js [hunk 2]
```diff
@@ -63,22 +63,38 @@ InternalServerError.prototype = Object.create(HttpError.prototype);
...
+      break;
   }
+
+  return response
+    .json()
+    .then(body => {
+      error.message = body.error_description;
+      throw error;
+    })
+    .catch(err => {
```
Comment (Fe4nn0r) [inline src/Error.js]: Isn't it pointless to catch an error to throw it just after ?
